File: 414-ThirdMaximumNumber/414-ThirdMaximumNumber.py
# Last updated: 8/21/2025, 12:09:37 PM
class Solution(object):
    def thirdMax(self, n):
        """
        :type nums: List[int]
        :rtype: int
        """
        l=[]
        #n=sorted (n)
        #n=n[::-1]
        '''for i in sorted(n)[::-1]:
            if i not in l:
                l.append(i)
        if len(l)>2:
            return l[2]
        else:
            return max(l)'''
        for i in set(n):
            l.append(i)
        l=sorted(l)[::-1]
        if len(l)>2:
            return l[2]
        else:
            return max(l)

        

File: 414-ThirdMaximumNumber/414-ThirdMaximumNumber.java
// Last updated: 8/21/2025, 12:09:31 PM
class Solution {
    public int thirdMax(int[] nums) {
        int c=0;
        Arrays.sort(nums);
        int res=nums[nums.length-1];
        for(int i=nums.length-2;i>=0;i--){
            if(nums[i]<res){
                res=nums[i];
                c++;
            }
            if(c==2){
                return nums[i];
            }
        }
        return nums[nums.length-1];
        
        
    }
}